ext/liquid_c/vm_assembler.h

```diff
@@ -42,7 +42,7 @@ static inline void vm_assembler_add_leave(vm_assembler_t *code)
 
 static inline void vm_assembler_remove_leave(vm_assembler_t *code)
 {
-    code->instructions.size -= 1;
+    code->instructions.data_end--;
```

Comment:
Instead of just decrementing the counter, should we have a function that decrements the size of the buffer which could also realloc and free some memory if we remove beyond a certain threshold?

Reply:
This would be a bad place to shrink the buffer, since the leave instruction is being removed in order to extend the buffer after which the leave instruction will be added again.